xla/service/gpu/parallel_loop_emitter.cc

```diff
@@ -184,7 +184,11 @@ ParallelLoopEmitter::EmitIndexAndSetExitBasicBlock(absl::string_view loop_name,
   // in that cases.
   llvm::Value* row_index = nullptr;
   if (!launch_config_.row_vectorized) {
-    array_indices.emplace_back(linear_index_base, shape_, b_);
+    llvm::Value* linear_index =
```

Comment:
Updated with my comment.
```cpp
    // The add operation is needed even if the offset is 0, since when the
    // kernel is unrolled, the following GEP instruction shares the same pointer
    // and sequential indices with others, allowing the default SLP pass to
    // optimize them into vectorized load/store operations.
```